chainer/testing/link.py

```diff
@@ -0,0 +1,549 @@
+            raise unittest.SkipTest('skip_backward_test is set')
+
+        # avoid cyclic import
+        from chainer import gradient_check
+
+        def do_check():
+            self.before_test('test_backward')
+
+            inits = self._generate_forward_backward_initializers()
+            link = self._create_initialized_link(inits, backend_config)
+
+            def f(inputs, ps):
+                with link.init_scope():
+                    for param_name, p in zip(self.param_names, ps):
+                        setattr(link, param_name, p)
+                return self._forward(link, inputs, backend_config)
+
+            link, inputs, outputs = self._create_initialized_link(
+                inits, backend_config, return_inputs_outputs=True)
+
+            params = _get_link_params(link, self.param_names)
+            params = [p.array for p in params]
+
+            cpu_device = backend.CpuDevice()
+            outputs = [cpu_device.send(output) for output in outputs]
+            grad_outputs = self._generate_grad_outputs(outputs)
+            grad_outputs = backend_config.get_array(grad_outputs)
+
+            inputs = self._to_noncontiguous_as_needed(inputs)
+            params = self._to_noncontiguous_as_needed(params)
+            grad_outputs = self._to_noncontiguous_as_needed(grad_outputs)
+
+            with LinkTestError.raise_if_fail(
+                    'backward is not implemented correctly'):
+                gradient_check._check_backward_with_params(
+                    f, inputs, grad_outputs, params=params,
+                    dtype=numpy.float64,
+                    detect_nondifferentiable=self.dodge_nondifferentiable,
+                    **self.check_backward_options)
+
+        if self.dodge_nondifferentiable:
+            while True:
+                try:
+                    do_check()
+                except gradient_check.NondifferentiableError:
+                    continue
+                else:
+                    break
+        else:
+            do_check()
+
+    def test_initializers(self, backend_config):
+        """Tests that the parameters of a links are correctly initialized."""
+        if self.skip_initializers_test:
+            raise unittest.SkipTest('skip_initializers_test is set')
+
+        self.before_test('test_initializers')
+
+        params_inits = self._generate_initializers()
+
+        default_init = self.default_initializer
+        initializers._check_is_initializer_like(default_init)
+
+        for i_param, param_inits in enumerate(params_inits):
+            # When testing an initializer for a particular parameter, other
+            # initializers are set to the default initializer.
+
+            inits = [default_init, ] * len(params_inits)
+
+            for init in param_inits:
+                inits[i_param] = init
+                self._test_single_initializer(i_param, inits, backend_config)
+
+    def _test_single_initializer(self, i_param, inits, backend_config):
```

Comment:
Write comments about logic (before each block).

Reply:
Fixed. Can you check if it's what you mean?